meilisearch-lib/src/tasks/scheduler.rs

```diff
@@ -82,29 +83,38 @@ impl DerefMut for TaskList {
 }
 
 impl TaskList {
-    fn new(index: String) -> Self {
+    fn new(id: TaskListIdentifier) -> Self {
         Self {
-            index,
+            id,
             tasks: Default::default(),
         }
     }
 }
 
 impl PartialEq for TaskList {
     fn eq(&self, other: &Self) -> bool {
-        self.index == other.index
+        self.id == other.id
     }
 }
 
 impl Eq for TaskList {}
 
 impl Ord for TaskList {
     fn cmp(&self, other: &Self) -> Ordering {
-        match (self.peek(), other.peek()) {
-            (None, None) => Ordering::Equal,
-            (None, Some(_)) => Ordering::Less,
-            (Some(_), None) => Ordering::Greater,
-            (Some(lhs), Some(rhs)) => lhs.cmp(rhs),
+        match (&self.id, &other.id) {
+            (TaskListIdentifier::Index(_), TaskListIdentifier::Index(_)) => {
+                match (self.peek(), other.peek()) {
+                    (None, None) => Ordering::Equal,
+                    (None, Some(_)) => Ordering::Less,
+                    (Some(_), None) => Ordering::Greater,
+                    (Some(lhs), Some(rhs)) => lhs.cmp(rhs),
+                }
+            }
+            (TaskListIdentifier::Index(_), TaskListIdentifier::Dump) => Ordering::Less,
+            (TaskListIdentifier::Dump, TaskListIdentifier::Index(_)) => Ordering::Greater,
+            (TaskListIdentifier::Dump, TaskListIdentifier::Dump) => {
+                unreachable!("There should be only one Dump task list")
```

Comment:
I am not sure that I understand, does it mean that there can only be one dump task in the task list? That a user cannot ask for two dumps one after the other?

Reply:
no, this is not what is happening here. We create a specific task list for dump, that is identified with the `TaskListIdentifier::Dump` identifier. If the user asks for a new dump, it will be added to the existing dump list if it exists, or it will be created, but I can't have 2 task list with the `Dump` identifier

Reply:
> That a user cannot ask for two dumps one after the other?
Also, yes, that's a limitation we always had in meilisearch;
https://docs.meilisearch.com/reference/api/error_codes.html#error-list